### ISSUE TITLE
CSGCombiner3D is modified when changing scenes with change_scene_to_file

### ISSUE THREAD

**Issue by Lippanon** [godotengine/godot#72771]
### Godot version

v4.0.beta.custom_build [2bd530270] ; v4.0.beta17.official [c40020513]

### System information

Windows 10, Vulkan

### Issue description

This is a regression since v4.0.beta16.official [518b9e580], though I haven't bisected the commit that caused it.

When using `change_scene_to_file` to load into a new scene, a CSGCombiner3D with a child CSGPolygon3D is modified from how it looks in the editor (or running the scene first).
Observe how the MRP scene looks in the editor:
![image](https://user-images.githubusercontent.com/115422160/216841352-d4976b77-3d62-4deb-8ee1-07d42a32eced.png)

How it looks when using Run Current Scene (F6):
![image](https://user-images.githubusercontent.com/115422160/216841407-396ea5cb-0490-499b-9a9a-732e645381ec.png)

How it looks when running the project with a simple script that loads into the same scene:
![image](https://user-images.githubusercontent.com/115422160/216841451-161a877f-0f7d-4f76-99b2-b21f6f20fddb.png)

The script:
```gdscript
extends Control

func _ready() -> void:
	get_tree().change_scene_to_file("res://node_3d.tscn") # res://node_3d.scn is the same scene but it wouldn't open in Beta 16 that I used to test (corrupt)
```

The hole is gone. The remote editor seems to show the same values and the nodes are all there.
I don't know if something else broke with `change_scene_to_file`, if someone knows a workaround please share it.

### Steps to reproduce

Run the MRP. Note: the 3d scene .scn binary format won't open in Beta 16, the .tscn is the same scene and it opens, that might be a clue.

### Minimal reproduction project

[csg bug.zip](https://github.com/godotengine/godot/files/10612034/csg.bug.zip)

**Comment by Lippanon** [comment]
I used Windows Builds artifacts to narrow it down (it takes me a long time to build Godot):
v4.0.beta.custom_build [045822e66] - no bug
v4.0.beta.custom_build [f5cf2d992] - no bug
v4.0.beta.custom_build [5bcf016f1] - has bug

I don't know if this necessarily means that https://github.com/godotengine/godot/pull/62737 introduced it.
Reference: https://github.com/godotengine/godot/actions/workflows/windows_builds.yml?page=19

**Comment by akien-mga** [comment]
To properly bisect using GitHub artifacts, you should use only artifacts from "Merge pull request" commits. The other ones are from PRs, and may be based on older branches and not include what caused the regression (even if the build happened later).

**Comment by Lippanon** [comment]
v4.0.beta.custom_build [62689affc] - has bug
v4.0.beta.custom_build [eedc4553b] - no bug

Which should mean https://github.com/godotengine/godot/pull/72427 has caused this regression. @MinusKube

**Comment by etodd** [comment]
I am running into this too, and I found that everything works when collision is enabled. Based on the original bug #71240 that was fixed by #72427, I think this should be an easy fix. I'll attempt it.

**Comment by etodd** [comment]
My PR (#77118) fixes this:

<img width="615" alt="image" src="https://github.com/godotengine/godot/assets/1226339/13e36289-70d3-4928-854b-8d33d8fcc63f">
